src/test/java/fakes/ThrowingRelpConnectionFake.java

```diff
@@ -0,0 +1,126 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+package fakes;
+
+import com.teragrep.lsh_01.config.RelpConfig;
+import com.teragrep.lsh_01.pool.IRelpConnection;
+import com.teragrep.rlp_01.RelpBatch;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Fake that throws an exception when connecting until the amount of connect attempts reach the given limit.
+ */
+public final class ThrowingRelpConnectionFake implements IRelpConnection {
+
+    final private IRelpConnection relpConnection;
+    final private int limit;
+    private int timesConnected;
+
+    public ThrowingRelpConnectionFake(IRelpConnection relpConnection, int limit) {
+        this.relpConnection = relpConnection;
+        this.limit = limit;
+    }
+
+    @Override
+    public int getReadTimeout() {
+        return relpConnection.getReadTimeout();
+    }
+
+    @Override
+    public void setReadTimeout(int readTimeout) {
+        relpConnection.setReadTimeout(readTimeout);
+    }
+
+    @Override
+    public int getWriteTimeout() {
+        return relpConnection.getWriteTimeout();
+    }
+
+    @Override
+    public void setWriteTimeout(int writeTimeout) {
+        relpConnection.setWriteTimeout(writeTimeout);
+    }
+
+    @Override
+    public int getConnectionTimeout() {
+        return relpConnection.getConnectionTimeout();
+    }
+
+    @Override
+    public void setConnectionTimeout(int timeout) {
+        relpConnection.setConnectionTimeout(timeout);
+    }
+
+    @Override
+    public void setKeepAlive(boolean on) {
+        relpConnection.setKeepAlive(on);
+    }
+
+    @Override
+    public int getRxBufferSize() {
+        return relpConnection.getRxBufferSize();
+    }
+
+    @Override
+    public void setRxBufferSize(int size) {
+        relpConnection.setRxBufferSize(size);
+    }
+
+    @Override
+    public int getTxBufferSize() {
+        return relpConnection.getTxBufferSize();
+    }
+
+    @Override
+    public void setTxBufferSize(int size) {
+        relpConnection.setTxBufferSize(size);
+    }
+
+    @Override
+    public boolean connect(String hostname, int port) throws IOException, TimeoutException {
+        timesConnected++;
+        if (timesConnected <= limit) {
+            throw new IOException("Fake exception");
+        }
+        return relpConnection.connect(hostname, port);
+    }
+
+    @Override
+    public void tearDown() {
+        relpConnection.tearDown();
+    }
+
+    @Override
+    public boolean disconnect() throws IOException, IllegalStateException, TimeoutException {
+        return relpConnection.disconnect();
+    }
+
+    @Override
+    public void commit(RelpBatch relpBatch) throws IOException, IllegalStateException, TimeoutException {
+        relpConnection.commit(relpBatch);
```

Comment:
should there be a test where commit throws too?

Reply:
A corresponding one is in the ResendingRelpConnectionFake, which has the commit as a no-op, resulting in a resend. An exception being thrown would result in the same behavior.